overlord/devicestate/remodel.go

```diff
@@ -238,6 +254,38 @@ func (rc *baseRemodelContext) SystemMode() string {
 	return rc.systemMode
 }
 
+func (rc *baseRemodelContext) setRecoverySystemLabel(label string) {
+	rc.recoverySystemLabel = label
+}
+
+func (rc *baseRemodelContext) postFinish() error {
```

Comment:
I think to think a bit about a suggestion but this is not a great name

Reply:
thinking of similar helpers elsewhere, maybe it should be called:

maybeUpdateBoot

with a doc comment explaining what it does